### DISCUSSION TITLE
entangle fallback value

### DISCUSSION THREAD

**Discussion by nam-co** [livewire/livewire#4182] [1. Help] [answered]
Hi, I have a this situation, where livewire eliminates the array and @entangle is actually faster than the php re-render

```
x-data="{ subItems_{{$itemData['id']}}: @entangle('itemsArray.'.$key.'.files')

<template x-for="(subItem, index) in subItems_{{$itemData['id']}}" :key="Date.now()+subItem.id">
```

this produces this error because 'itemsArray.'.$key.'.files' no longer exist (undefined)
```
Expression: "subItems_xxx" undefined is not an object (evaluating 'n.length')
```

Appreciate any any ideas for a way to fix this? maybe by pasing the entangle trough a function? where if !isset returns a blank array? thanks

I wish there was a condition fallback value option for @entangle($something, []) if value == 'undefined' return fallback

**Comment by godismyjudge95** [comment]
If you want to fallback to a default value you can just use the nullish coalescing operator `??` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

It would look like this in your example:
```
x-data="{ subItems_{{$itemData['id']}}: @entangle('itemsArray.'.$key.'.files') ?? []
```

Then whenever the `'itemsArray.'.$key.'.files'` doesn't exist it will fall back to the empty array.

**Reply by nam-co** [reply]
Hi @godismyjudge95 thanks for the response, I tried that but it didn't work, but I finally got it working by using x-for="(subItem, index) in (subItems_{{$itemData['x_id']}} ?? [])"

**Reply by nam-co** [reply]
anyway it could be nice to have a fallback option in @entangle($var, $fallback) @calebporzio

**Comment by nam-co** [comment, marked as answer]
Hi @godismyjudge95 thanks for the response, I tried that but it didn't work, but I finally got it working by using x-for="(subItem, index) in (subItems_{{$itemData['x_id']}} ?? [])"
